### PR TITLE
[Fix issue #2150] Remove waiting in Ray init

### PR DESCRIPTION
**Pull request by yangw1234**
## Current Problem
Currently, `RayContext.init()` will wait at least 13 seconds before start. The current flow is
1. if `task_id == 0`, first sleep 5 seconds, then start ray master, then sleep 8 seconds.
2. if `task_id != 0`, first sleep 8 seconds, then start ray slaves, then sleep 5 seconds.

So, there is a 3 seconds gap to make sure ray slaves is started after ray master is started. 
This 3 seconds gay might be too long for a small cluster and might not be enough for a large cluster. And there are also 10 seconds sleep time does not seem to be useful.

## Proposed Solution
We can put a barrier between master start and slave start, e.g.
```python
def map_func(iter):
    if task_id == 0:
        start_ray_master()
    tc.barrier()
    if task_id != 0:
        start_ray_slave()
```

## Testing
1. existing unit tests
2. manually tested it on yarn cluster

### CODE REVIEW

**Comment by shane-huang** [comment]
I think barrier should be okay for this.  And maybe we should add some intergation tests (on cluster) for this test.

**Comment by hkvision** [comment]
LGTM

File: pyzoo/zoo/ray/util/raycontext.py [hunk 1]
```diff
@@ -136,7 +135,7 @@ def _get_raylet_command(redis_address,
                                                        object_store_memory=object_store_memory,
                                                        extra_params=extra_params)
 
-    def _start_ray_node(self, command, tag, wait_before=5, wait_after=5):
+    def _start_ray_node(self, command, tag, wait_before=0, wait_after=0):
```
Comment (jason-dai) [inline pyzoo/zoo/ray/util/raycontext.py:138]: just remove wait_before and wait_after?

Comment (yangw1234) [reply pyzoo/zoo/ray/util/raycontext.py]: removed

**Comment by jason-dai** [comment]
LGTM

**Comment by yangw1234** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/3166/ jenkins passes, except one failure caused by network issue.

**Pull request merged**